src/datanode/src/instance/grpc.rs

```diff
@@ -27,14 +27,28 @@ use substrait::{DFLogicalSubstraitConvertor, SubstraitPlan};
 use table::requests::CreateDatabaseRequest;
 use tonic::Request;
 
-use crate::error::{
-    DecodeLogicalPlanSnafu, ExecuteSqlSnafu, FlightGetSnafu, InvalidFlightDataSnafu, Result,
-};
+use crate::error::{DecodeLogicalPlanSnafu, ExecuteSqlSnafu, InvalidFlightDataSnafu, Result};
 use crate::instance::Instance;
 
 impl Instance {
     async fn boarding(&self, ticket: Request<Ticket>) -> Result<ObjectResult> {
-        let response = self.do_get(ticket).await.context(FlightGetSnafu)?;
+        let response = self.do_get(ticket).await;
+        let response = match response {
+            Ok(response) => response,
+            Err(e) => {
+                // TODO(LFC): Further formalize error message when Arrow Flight adoption is done,
+                // and don't forget to change "test runner"'s error msg accordingly.
+                return Ok(ObjectResult {
+                    header: Some(ResultHeader {
+                        version: 1,
+                        code: StatusCode::Internal as _,
```

Comment:
Is it ok to overwrite error code?

Reply:
I didn't bother to get the inner error's error code here, because I didn't see a meaningful usage for error code for now, and the code here is not stable yet (could be changed in the future).

Reply:
The error code is a part of UI/UX, it's better to display it if we can do it here.